tracer/src/Datadog.Trace/Tracer.cs

```diff
@@ -145,6 +145,11 @@ public static Tracer Instance
                       " instance in code.")]
             set
             {
+                if (value is null)
+                {
+                    ThrowHelper.ThrowInvalidOperationException("The tracer instance cannot be set to null.");
```

Comment:
```suggestion
                    ThrowHelper.ThrowInvalidOperationException("The tracer instance shouldn't be set to null as this would cause issues to the automatic instrumentation.");
```